### ISSUE TITLE
DateTime rule not validating dates correctly

### ISSUE THREAD

**Issue by NickBelhomme** [particle-php/Validator#31]
# Context

When passing a format to the datetime rule, it even accepts dates which are in another format because of the default behavior of the \DateTime implementation.

# Use Case
```php
$validator->required('date')->datetime('Ymd');
$validator->validate(['date' => '12111978']); // 12 november 1978
```
The above will give true while 19 can never be a correct month, nor can 78 be a correct day.

The issue lies within this part of the code:

```php
$date = date_create_from_format('Ymd', '11121978');
var_dump($date);
```

which gives:
```
object(DateTime)#1 (3) {
  ["date"]=>
  string(26) "1212-09-16 06:48:11.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(10) "US/Pacific"
}
```
https://github.com/particle-php/Validator/blob/master/src/Particle/Validator/Rule/Datetime.php#L75

# Acceptance Criteria
When issuing a validation using the datetime rule, it must adhere to the format AND also verify if the date portions are the same. This ensures that  the overflow is not used and dates such as above and 
31-02-2015 are considered invalid.

**Comment by berry-langerak** [comment]
That is a very excellent point, @NickBelhomme. I tend to forget this auto behaviour of `\DateTime`. Even though it can be helpful at times (start bill cycle on 29th of February == 1st of March), the functionality is not that useful when validating ;)

Will implement a fix soon!

**Comment by RickvdStaaij** [comment]
This bug is solved in #35 and is in version 0.4.0, closing.
